compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1149,6 +1149,24 @@ OMR::Z::TreeEvaluator::returnEvaluator(TR::Node * node, TR::CodeGenerator * cg)
          comp->setReturnInfo(TR_ObjectReturn);
          dependencies->addPostCondition(returnValRegister, linkage->getIntegerReturnRegister());
          break;
+      case TR::breturn:
+         dependencies->addPostCondition(returnValRegister, linkage->getIntegerReturnRegister());
+         comp->setReturnInfo(TR_ByteReturn);
+         if (linkage->isNeedsWidening())
+            {
+            /* LGBR equiv to {Operands R_1,R_2 : Load Byte (64<-8)} */
+            new (cg->trHeapMemory()) TR::S390RRInstruction(TR::InstOpCode::LGBR,
```

Comment:
Please avoid using placement new. We have helpers which may perform extra checks via `generateRRInstruction`. There are many examples of uses of this API throughout the Z codegen.